Restructure docs (#374)
kyma-project/application-connector-manager: docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md

# Configuring the Runtime

Runtime Agent periodically requests for the configuration of Kyma runtime from Unified Customer Landscape (UCL).

To fetch the Kyma runtime configuration, Runtime Agent calls the [`applicationsForRuntime`](https://github.com/kyma-incubator/compass/blob/master/components/director/pkg/graphql/schema.graphql) query offered by the component called UCL Director.

> The response for the query contains Applications (an `Application` represents an external system) assigned for Kyma runtime.

Reviewer: ```suggestion
The response for the query contains Applications (an Application represents an external system) assigned for Kyma runtime.
```

> Each `Application` contains credentials that are only valid and unique for the querying Kyma runtime (another requestor cannot use them). Runtime Agent stores the credentials in a Secrets which will be used by Application Gateway for establishing a trusted outbound communication to an external system.

Reviewer: ```suggestion
Each Application contains credentials that are only valid and unique for querying Kyma runtime (another requestor cannot use them). Runtime Agent stores the credentials in Secrets, which are used by Application Gateway to establish a trusted outbound communication to an external system.
```

> This data mapping shows how the retrieved configuration of an `Application` from UCL Director is stored in Kyma runtime:

Reviewer: ```suggestion
This data mapping shows how the retrieved configuration of an Application from UCL Director is stored in Kyma runtime:
```

| **UCL Director**    | **Kyma Runtime**                    |
|---------------------------|-------------------------------|
| Application               | Application CR                |
| API Bundle                | Service in the Application CR |
| API Definition            | Entry under the service       |

## Application Name

> The name of the `Application` is used as a key within the Application Connector module and has special requirements:

Reviewer: ```suggestion
The name of the Application is used as a key within the Application Connector module and has the following special requirements:
```

# Configuring the Runtime

Runtime Agent periodically requests for the configuration of Kyma runtime from Unified Customer Landscape (UCL).

> To fetch the Kyma runtime configuration, Runtime Agent calls the [`applicationsForRuntime`](https://github.com/kyma-incubator/compass/blob/master/components/director/pkg/graphql/schema.graphql) query offered by the component called UCL Director.

Reviewer: ```suggestion
To fetch the Kyma runtime configuration, Runtime Agent calls the [`applicationsForRuntime`](https://github.com/kyma-incubator/compass/blob/master/components/director/pkg/graphql/schema.graphql) query offered by the component called the UCL Director.
```

The response for the query contains Applications (an Application represents an external system) assigned for Kyma runtime.

Each Application contains credentials that are only valid and unique for querying Kyma runtime (another requestor cannot use them). Runtime Agent stores the credentials in Secrets, which are used by Application Gateway to establish a trusted outbound communication to an external system.

> This data mapping shows how the retrieved configuration of an Application from UCL Director is stored in Kyma runtime:

Reviewer: ```suggestion
This data mapping shows how the retrieved configuration of an Application from the UCL Director is stored in Kyma runtime:
```

| **UCL Director**    | **Kyma Runtime**                    |
|---------------------------|-------------------------------|
| Application               | Application CR                |
| API Bundle                | Service in the Application CR |
| API Definition            | Entry under the service       |

## Application Name

The name of the Application is used as a key within the Application Connector module and has the following special requirements:

### Uniqueness

The names of Applications assigned to the Runtime must be unique in Kyma Runtime. If they are not unique, the synchronization fails.

### Normalization of Application Names

> Runtime Agent can normalize the names of Applications fetched from UCL Director by converting them to lowercase and removing special characters and spaces.

Reviewer: ```suggestion
Runtime Agent can normalize the names of Applications fetched from the UCL Director by converting them to lowercase and removing special characters and spaces.
```
